joint_limits/include/joint_limits/joint_limits_rosparam.hpp

```diff
@@ -14,17 +14,84 @@
 
 /// \author Adolfo Rodriguez Tsouroukdissian
 
-#ifndef JOINT_LIMITS_INTERFACE__JOINT_LIMITS_ROSPARAM_HPP_
-#define JOINT_LIMITS_INTERFACE__JOINT_LIMITS_ROSPARAM_HPP_
+#ifndef JOINT_LIMITS__JOINT_LIMITS_ROSPARAM_HPP_
+#define JOINT_LIMITS__JOINT_LIMITS_ROSPARAM_HPP_
 
-#include <joint_limits_interface/joint_limits.hpp>
+#include <limits>
+#include <string>
 
-#include <rclcpp/rclcpp.hpp>
+#include "joint_limits/joint_limits.hpp"
+#include "rclcpp/rclcpp.hpp"
 
-#include <string>
+namespace  // utilities
+{
+/// Declare and initialize a parameter with a type.
+/**
+   *
+   * Wrapper function for templated node's declare_parameter() which checks if
+   * parameter is already declared.
+   * For use in all components that inherit from ControllerInterface
+   */
+template <typename ParameterT>
+auto auto_declare(
+  const rclcpp::Node::SharedPtr & node, const std::string & name, const ParameterT & default_value)
+{
+  if (!node->has_parameter(name))
+  {
+    return node->declare_parameter<ParameterT>(name, default_value);
+  }
+  else
+  {
+    return node->get_parameter(name).get_value<ParameterT>();
+  }
+}
+}  // namespace
 
-namespace joint_limits_interface
+namespace joint_limits
+{
+inline bool declare_parameters(const std::string & joint_name, const rclcpp::Node::SharedPtr & node)
```

Comment:
Quick issue I discovered when I tried to use this package:

The controllers are now `rclcpp_lifecycle::LifecycleNode`. Which means, I believe, they can't use this function.

Can you add another version of this function that takes a LifecycleNode:

`inline bool declare_parameters(const std::string & joint_name, const rclcpp_lifecycle::LifecycleNode::SharedPtr & node)`

Reply:
Or, `.automatically_declare_parameters_from_overrides` is already set to `true` so maybe this function isn't even needed.

Reply:
You are fully correct!!! Silly me… not the updated version. Thanks for checking this so thoroughly!